src/m365/spe/commands/container/container-list.ts

```diff
@@ -0,0 +1,113 @@
+import { Logger } from '../../../../cli/Logger.js';
+import GlobalOptions from '../../../../GlobalOptions.js';
+import { formatting } from '../../../../utils/formatting.js';
+import { odata } from '../../../../utils/odata.js';
+import { validation } from '../../../../utils/validation.js';
+import GraphCommand from '../../../base/GraphCommand.js';
+import commands from '../../commands.js';
+import { ContainerProperties } from '../../ContainerProperties.js';
+import { ContainerTypeProperties, spo } from '../../../../utils/spo.js';
+
+interface CommandArgs {
+  options: Options;
+}
+
+export interface Options extends GlobalOptions {
+  containerTypeId?: string;
+  containerTypeName?: string;
+}
+
+class SpeContainerListCommand extends GraphCommand {
+  public get name(): string {
+    return commands.CONTAINER_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists all Container Types';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'displayName', 'containerTypeId', 'createdDateTime'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        containerTypeId: typeof args.options.containerTypeId !== 'undefined',
+        containerTypeName: typeof args.options.containerTypeName !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '--containerTypeId [containerTypeId]'
+      },
+      {
+        option: '--containerTypeName [containerTypeName]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.containerTypeId && !validation.isValidGuid(args.options.containerTypeId as string)) {
+          return `${args.options.containerTypeId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['containerTypeId', 'containerTypeName'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      if (this.verbose) {
+        await logger.logToStderr(`Retrieving list of Containers...`);
+      }
+
+      const containerTypeId = await this.getContainerTypeId(logger, args);
```

Comment:
lets try to avoid passing `args` but instead just pass `args.options` as good practice